### ISSUE TITLE
nil pointer evaluating interface {}.Name when serviceAccount is defined for beats

### ISSUE THREAD

**Issue by andrewnazarov** [elastic/cloud-on-k8s#6904]
## Bug Report
**What did you do?**
We wanted to install Metricbeat via the eck-stack chart. We disabled all other components and enabled beats only, then defined a configuration for Metricbeat specifically. When we defined a serviceAccount to have it created the error appeared.

**What did you expect to see?**
Installed Metricbeat with all necessary permissions.

**What did you see instead? Under which circumstances?**
The following error appeared:
```
Error: template: eck-stack/charts/eck-beats/templates/service-account.yaml:10:8: executing "eck-stack/charts/eck-beats/templates/service-account.yaml" at <include "beat.labels" .>: error calling include: template: eck-stack/charts/eck-beats/templates/_helpers.tpl:37:18: executing "beat.labels" at <include "beat.chart" .>: error calling include: template: eck-stack/charts/eck-beats/templates/_helpers.tpl:30:25: executing "beat.chart" at <.Chart.Name>: nil pointer evaluating interface {}.Name
```

**Environment**

* ECK version: chart version 0.5.0

* Kubernetes information: GKE 1.24

**How to reproduce**

The easy way to reproduce this is to run the following. There is even no need for disabling other components.

```
helm template --set eck-beats.enabled=true --set eck-beats.spec.type=metricbeat --set eck-beats.spec.daemonSet.podTemplate.spec.serviceAccountName=metricbeat --set eck-beats.serviceAccount.name=metricbeat my-eck-stack elastic/eck-stack --version 0.5.0
```

**Comment by hadams95** [comment]
@andrewnazarov

Hit this today as well and I think I have identified the problem at the following line in `eck-beats` **service-account.yaml**
https://github.com/elastic/cloud-on-k8s/blob/main/deploy/eck-stack/charts/eck-beats/templates/service-account.yaml#L2

It is using `with` on the `serviceAccount` block.  Inside of that, it is also calling a helper function `beat.labels` and passing parameter `.`

Because of the `with`, the value of `.` passed into the helper function is actually scoped to the block `serviceAccount` rather than the entire values interface for the eck-beats subchart.  Thus the helper function is actually attempting to access `eck-beats.serviceAccount.Chart.Name` rather than just `.Chart.Name`.

I was able to eliminate the error by adjusting my `serviceAccount` block to something like:
```yaml
eck-beats:
  serviceAccount:
    name: some-name
    Chart:
      Name: some-chart-name
      Version: some-chart-version
    Values:
      nameOverride: some-name-override
    Release:
      Name: some-release-name
      Service: some-service-name
```

This is obviously bad experience and likely unintended and/or a bug with the chart.
